query-connector/src/app/query/components/CustomizeQuery.tsx

```diff
@@ -99,6 +112,9 @@ const CustomizeQuery: React.FC<CustomizeQueryProps> = ({
       (item) => ({
         ...item,
         include: checked, // Set all items in this group to checked or unchecked
```

Comment:
I think as it currently is this needs to be `includeValueSet` in order for this to work properly, since we have a different include type bool at the valueset vs concept level. In line 88 of `queryHooks`, we're checking `includeValueSet`, not include, which I think is causing the filtering to break.

In my mind though there's no reason we need different properties at the two levels: we should have them be the same to reduce the cognitive complexity of what consitutes "include". Kknowing it's quite a bit of work (😬), what do we think of changing this at the object level rather than having to remember the difference between the two?

Reply:
Relatedly, I think we need to add an e2e to check the whole customize query flow since that's like, a core part of the current app. Made a ticket for that [here](https://app.zenhub.com/workspaces/skylight-dibbs-viper-6480ba23b5a07644e0e46d23/issues/gh/cdcgov/dibbs-query-connector/44)

Reply:
Could you elaborate on what you mean by the filtering is breaking? Visually, everything looks like it's working as expected so not really sure what you mean.

Re: two levels of include: that's something Marcelle put in her spike work, that both `Concept` and `ValueSet` should have include properties. I can't seem to find the reasoning for it, but given the depth she went on her research I'd probably prefer to leave this as-is, at least until she's back. Given that we're also going to have a discussion Thursday on DB schema, I would likely opt to not change the include structure in this PR, especially since it was ~3 days of work and I'd like to get it merged soon so we can capture the changes and keep everything up to date. Conceptually, I think the idea is that we include a value set if at least one concept in that value set is included in a query, so if that is the rationale then `includeValueSet` becomes a convenience property rather than something strictly "necessary."

Reply:
Given my theoretical explanation of the rationale for two includes, I think the right approach would actually be to set both properties to checked.

Reply:
LOL ignore everything I just wrote then

Reply:
Ah excellent, that helps a lot. I'll dig into this and see if changing the includes helps. Thanks!